libp2p/dialer.nim

```diff
@@ -179,32 +173,30 @@ proc internalConnect(
     await lock.acquire()
 
     if peerId.isSome and reuseConnection:
-      let connOpt = await self.tryReusingConnection(peerId.get())
-      if connOpt.isSome:
-        return connOpt.get()
+      let muxOpt = await self.tryReusingConnection(peerId.get())
+      if muxOpt.isSome:
+        return muxOpt.get()
 
     let slot = self.connManager.getOutgoingSlot(forceDial)
-    let conn =
+    let muxed =
       try:
         await self.dialAndUpgrade(peerId, addrs)
       except CatchableError as exc:
         slot.release()
         raise exc
-    slot.trackConnection(conn)
-    if isNil(conn): # None of the addresses connected
+    slot.trackMuxer(muxed)
+    if isNil(muxed): # None of the addresses connected
       raise newException(DialFailedError, "Unable to establish outgoing link")
 
-    # A disconnect could have happened right after
-    # we've added the connection so we check again
-    # to prevent races due to that.
-    if conn.closed() or conn.atEof():
-      # This can happen when the other ends drops us
-      # before we get a chance to return the connection
-      # back to the dialer.
-      trace "Connection dead on arrival", conn
-      raise newLPStreamClosedError()
-
-    return conn
+    try:
+      await self.peerStore.identify(muxed)
```

Comment:
If we are adding the `peerStore` here only cause of `Identify`, could we move the `Identify` here instead?

Reply:
It's also used in the switch